streamlit_extras/echo_expander/__init__.py

```diff
@@ -0,0 +1,143 @@
+import streamlit as st
+# import contextlib
+# from streamlit.echo import ( _get_initial_indent, _get_indent )
+# from typing import List
+# import textwrap
+# import traceback
+
+# @contextlib.contextmanager
+# def echo_expander(title="Show code"):
+#     from streamlit import code, warning, empty, source_util, expander
+#     frame = traceback.extract_stack()[-3]
+#     filename, start_line = frame.filename, frame.lineno
+#     with source_util.open_python_file(filename) as source_file:
+#         source_lines = source_file.readlines()
+#     initial_indent = _get_initial_indent(source_lines[start_line:])
+#     lines_to_display: List[str] = []
+#     for line in source_lines[start_line:]:
+#         indent = _get_indent(line)
+#         if indent is not None and indent < initial_indent:
+#             break
+#         lines_to_display.append(line)
+#     code_string = textwrap.dedent("".join(lines_to_display))
+#     yield
+#     expander(title).code(code_string, "python")
+#     st.markdown('---')
+
+###
+# Extension from echo() in streamlit/echo.py
+###
+
+import contextlib
+import textwrap
+import traceback
+from typing import List
+
+@contextlib.contextmanager
+def echo_expander(code_location="above", expander=True, label="Show code"):
```

Comment:
Should `expander` be an option? It seems like expander=False would be a strange use-case, when you could just use st.echo.

Reply:
I agree, however, I was sort of thinking that echo_expander could become a complete replacement for echo. I think the Streamlit folks were hesitant to take this as a PR to change echo itself.
Honestly, I could go either way. It's simple enough to remove :)